src/projection_client.rs

```diff
@@ -0,0 +1,622 @@
+use crate::event_store::client::projections;
+use crate::event_store::client::shared::Empty;
+use crate::grpc::{ClientSettings, GrpcClient};
+use crate::options::projections::CreateProjectionOptions;
+use crate::Credentials;
+use futures::stream::BoxStream;
+use serde::de::DeserializeOwned;
+
+#[allow(dead_code)]
+#[derive(Clone, Debug)]
+pub(crate) enum StatsFor {
+    Name(String),
+    AllProjections,
+    AllContinuous,
+}
+
+#[derive(Clone, Debug)]
+pub struct ProjectionDetails {
+    pub core_processing_time: i64,
+    pub version: i64,
+    pub epoch: i64,
+    pub effective_name: String,
+    pub writes_in_progress: i32,
+    pub reads_in_progress: i32,
+    pub partitions_cached: i32,
+    pub status: String,
+    pub state_reason: String,
+    pub name: String,
+    pub mode: String,
+    pub position: String,
+    pub progress: f32,
+    pub last_checkpoint: String,
+    pub events_processed_after_restart: i64,
+    pub checkpoint_status: String,
+    pub buffered_events: i64,
+    pub write_pending_events_before_checkpoint: i32,
+    pub write_pending_events_after_checkpoint: i32,
+}
+
+#[derive(Clone)]
+pub struct ProjectionClient {
+    client: GrpcClient,
+    settings: ClientSettings,
+}
+
+impl ProjectionClient {
+    pub async fn create(settings: ClientSettings) -> Result<Self, Box<dyn std::error::Error>> {
+        let client = GrpcClient::create(settings.clone()).await?;
+
+        Ok(ProjectionClient { client, settings })
+    }
+
+    pub async fn create_projection<Name>(
+        &self,
+        name: Name,
+        query: String,
+        options: &CreateProjectionOptions,
+    ) -> crate::Result<()>
+    where
+        Name: AsRef<str>,
+    {
+        self.create_projection_internal(
+            options.credentials.as_ref(),
+            projections::create_req::Options {
+                query: query.clone(),
+                mode: Some(projections::create_req::options::Mode::Continuous(
+                    projections::create_req::options::Continuous {
+                        name: name.as_ref().to_string(),
+                        track_emitted_streams: options.track_emitted_streams,
+                    },
+                )),
+            },
+        )
+        .await?;
+
+        // TODO - create projection RPC call needs to be fixed upstream where the emit options
+        // will be added to the API. Right now, do an extra RPC call to implement it.
+        if options.emit_enabled {
```

Comment:
**FIX:** I think that based on the RFC this code can be removed: https://github.com/EventStore/home/pull/524/files#diff-66701ffba6c5b4c0e1742dbaf2ff3780465f63d22820aa8ee280a30157f5c20aR16

Reply:
I disagree. The RFC says it's defaulted to `false`. The current gRPC protobuf definition doesn't expose that parameter at the create projection level. In order to not introduce a breaking change when that parameter will be added, I do that extra RPC call in the meantime.

Reply:
Right, my bad. There was a discussion about removing this option, but in the final RFC. It's still there 👍